### ISSUE TITLE
Error : Key not valid for use in specified state.

### ISSUE THREAD

**Issue by sharmagourav** [NuGet/NuGetGallery#2391]
I am using Artifactory as NuGet repository to store all the nuget packages. 
When we use this Artifactory link as source, the package Manager  in Visual Studio  prompts for credentials and worked very fine.
As soon as we moved to build machine (with out VS) and try to build application it is throwing error : 
.nuget\NuGet.targets(100,9): error : Key not valid for use in specified state.

I added source with -user -password and put config at local user location. I tried with ClearText password and encrypted password both are throwing same error. 

Am I missing anything here? Please advice.

![nuget_config](https://cloud.githubusercontent.com/assets/10505556/6925509/5a3e746c-d7a5-11e4-815f-61d3ee5a46a8.JPG)

**Comment by zhili1208** [comment]
what's the nuget version and visual studio version you are using?

**Comment by sharmagourav** [comment]
NuGet Package manager : 2.8.50926.663
Visual Studio 12.0

**Comment by zhili1208** [comment]
your config file looks right, Can you provide your NuGet.target file? I can't repro on my machine

**Comment by zhili1208** [comment]
I checked NuGet.target file, looks like line(100,9) is calling nuget restore command. I can't find where the error message come from. Can you use nuget.exe to restore your package in your build machine?
if there is any error, send the error message to me. thanks

**Comment by sharmagourav** [comment]
Thanks for reply zhili1208. Yes , I am able to restore everything using nugget.exe on build machine. But again it is restoring from nugget.org instead of Artifactory location.

**Comment by zhili1208** [comment]
that's good if nuget.org works for you. I can't figure out what's the issue from your description. If you still want to restore from Artifactory location, I can help.
First, make sure you can connect to Artifactory location from your build machine with your username and password, And use nuget config command to set user name and password, then try to use nuget.exe restore package from Artifactory location. if there is any error message, send to me. I can debug it. Thanks.

**Comment by sharmagourav** [comment]
Yup , I did the same thing as you mentioned.  I can connect to artifactory using my user name and Password from build server. I added them into nugget config using command line as shown in above image. 
But still it is restoring from nugget instead of artifactory.

Yes , I want to restore from Artifactory.

**Comment by zhili1208** [comment]
you can disable the nuget.org source, just left the artifactory source. Then can you restore package?

**Comment by sharmagourav** [comment]
then I got this error : .nuget\NuGet.targets(100,9): error : Key not valid for use in specified state.

**Comment by zhili1208** [comment]
for me, it looks like a CryptographicException exception. so if you use clearText password, it still has this error?

**Comment by sharmagourav** [comment]
yes. I tried with clearText too . it throws the same error.  Do need to check anything on artifactory side?

**Comment by zhili1208** [comment]
so if it still has this problem. I think this error comes from your server side. http client send the request to your server with the encoded username and password, but the server can't decode it. so got this error. Since I can't repro this on my side, I only can guess what's the issue.

**Comment by sstaszkiewicz-copperleaf** [comment]
Because googling `Key not valid for use in specified state` brought you here:

You might see this error when copying a nuget.config file with an encrypted password already stored in it, like into a windows container for example. Always run the `nuget sources add` or `nuget sources update` command on the same machine that you're planning to use nuget on.

After adding the nuget.config file to the container, try running `nuget sources update -Name Artifact -UserName %ARTIFACTORY_USER% -Password %ARTIFACTORY_API_KEY%` to fix up the nuget file.

**Comment by GersonDias** [comment]
Facing the same problem when copying a nuget.config file to a windows container to build it. If I need to run nuget sources add on it, what is the purpose of that file?

**Comment by Mattnificent** [comment]
I constantly have issues with nuget package feed authentication.  I have removed the source, and re-added it using `nuget sources add` with a brand new PAT, and arbitrary username (does not matter??? documentation is HIGHLY ambiguous...) - and I am still getting `Unable to load the service index for source <feed_url> Key not valid for use in specified state`.
Is someone at Microsoft turning off my access to my software that I have created?
